pkg/hvutils/hvutils.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hvutils
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+func GetInstanceTypeFromTags(tags []string, deviceId int32) (string, error) {
+	prefix := "instance-type="
+	var instanceTypes []string
```

Comment:
Use `instanceTypes := make([]string,0,1)`. It is a better practice as it makes clear that we don't expect anything more than one instanceType. In general, make should be used to avoid having to reallocate storage every time "append" is used. Here, it obviously doesn't matter, but still, we can stick to the "best practice".